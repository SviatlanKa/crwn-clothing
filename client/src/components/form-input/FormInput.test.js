import React from 'react';
import { shallow } from 'enzyme';
import FormInput from './FormInput';

describe('testing FormInput component', () => {
   let wrapper;
   let onHandleChange, label, otherProps;
   let mockProps;

   beforeAll(() => {
       onHandleChange = jest.fn();
       label = 'some text';
       otherProps = { value: ''};
       mockProps = { onHandleChange, label, ...otherProps };
       wrapper = shallow(<FormInput { ...mockProps } />);
   });

    it('render FormInput Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onChange when change in input', () => {
        wrapper.find('.form-input').simulate('change');
        expect(onHandleChange).toHaveBeenCalled();
    });

    it('check if label exists', () => {
        expect(wrapper.exists('label')).toBe(true);
    })
});