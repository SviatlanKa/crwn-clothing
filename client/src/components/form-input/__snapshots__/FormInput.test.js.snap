// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing FormInput component render FormInput Component 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    onChange={[MockFunction]}
    value=""
  />
  <label
    className="form-input-label"
  >
    some text
  </label>
</div>
`;
